crates/revm/src/instructions.rs

```diff
@@ -74,175 +72,165 @@ pub enum Return {
-        opcode::EXTCODECOPY => host::extcodecopy::<H, S>(interp, host),
-        opcode::RETURNDATASIZE => system::returndatasize::<S>(interp),
-        opcode::RETURNDATACOPY => system::returndatacopy::<S>(interp),
+        opcode::EXTCODESIZE => host::extcodesize::<S>(interp, host),
+        opcode::EXTCODEHASH => host::extcodehash::<S>(interp, host),
+        opcode::EXTCODECOPY => host::extcodecopy::<S>(interp, host),
+        opcode::RETURNDATASIZE => system::returndatasize::<S>(interp, host),
+        opcode::RETURNDATACOPY => system::returndatacopy::<S>(interp, host),
         opcode::BLOCKHASH => host::blockhash(interp, host),
         opcode::COINBASE => host_env::coinbase(interp, host),
         opcode::TIMESTAMP => host_env::timestamp(interp, host),
         opcode::NUMBER => host_env::number(interp, host),
         opcode::DIFFICULTY => host_env::difficulty::<H, S>(interp, host),
         opcode::GASLIMIT => host_env::gaslimit(interp, host),
-        opcode::SLOAD => host::sload::<H, S>(interp, host),
-        opcode::SSTORE => host::sstore::<H, S>(interp, host),
-        opcode::GAS => system::gas(interp),
-        opcode::LOG0 => host::log::<H, S>(interp, 0, host),
-        opcode::LOG1 => host::log::<H, S>(interp, 1, host),
-        opcode::LOG2 => host::log::<H, S>(interp, 2, host),
-        opcode::LOG3 => host::log::<H, S>(interp, 3, host),
-        opcode::LOG4 => host::log::<H, S>(interp, 4, host),
-        opcode::SELFDESTRUCT => host::selfdestruct::<H, S>(interp, host),
-        opcode::CREATE => host::create::<H, S>(interp, false, host), //check
-        opcode::CREATE2 => host::create::<H, S>(interp, true, host), //check
-        opcode::CALL => host::call::<H, S>(interp, CallScheme::Call, host), //check
-        opcode::CALLCODE => host::call::<H, S>(interp, CallScheme::CallCode, host), //check
-        opcode::DELEGATECALL => host::call::<H, S>(interp, CallScheme::DelegateCall, host), //check
-        opcode::STATICCALL => host::call::<H, S>(interp, CallScheme::StaticCall, host), //check
-        opcode::CHAINID => host_env::chainid::<H, S>(interp, host),
-        _ => Return::OpcodeNotFound,
+        opcode::SLOAD => host::sload::<S>(interp, host),
+        opcode::SSTORE => host::sstore::<S>(interp, host),
+        opcode::GAS => system::gas(interp, host),
+        opcode::LOG0 => host::log::<0, S>(interp, host),
+        opcode::LOG1 => host::log::<1, S>(interp, host),
+        opcode::LOG2 => host::log::<2, S>(interp, host),
+        opcode::LOG3 => host::log::<3, S>(interp, host),
+        opcode::LOG4 => host::log::<4, S>(interp, host),
+        opcode::SELFDESTRUCT => host::selfdestruct::<S>(interp, host),
+        opcode::CREATE => host::create::<false, S>(interp, host), //check
+        opcode::CREATE2 => host::create::<true, S>(interp, host), //check
+        opcode::CALL => host::call::<S>(interp, host),            //check
+        opcode::CALLCODE => host::call_code::<S>(interp, host),   //check
```

Comment:
Makes sense that the fn over enum is faster here

Reply:
For snailtracer it is not even called. Interpreter loop and maybe even `match` faster when it calls all same functions.